library/core/src/ptr/const_ptr.rs

```diff
@@ -106,6 +106,75 @@ impl<T: ?Sized> *const T {
         bits as Self
     }
 
+    /// Gets the "address" portion of the pointer.
+    ///
+    /// This is equivalent to `self as usize`, which semantically discards
+    /// *provenance* and *address-space* information. To properly restore that information,
+    /// use [`with_addr`][#method.with_addr] or [`map_addr`][#method.map_addr].
```

Comment:
Did `self::with_addr` not work here? I recall having issues with rustdoc accepting links to pointers, slices, arrays, and reperences, since those are simultaneously generic and unnameable without generics.

Reply:
...literally nothing in these files uses that syntax?

Reply:
like i'm not saying you're wrong just what the fuck is with this file

Reply:
I assume it's just untouched since proper path linking was implemented, *or* I was right about pointer linking being broken and you just didn't whisper the right dark magic incantation

Reply:
lmao:

```
error: unresolved link to `self::with_addr`
   --> library/core/src/ptr/const_ptr.rs:113:27
    |
113 |     /// use [`with_addr`][self::with_addr] or [`map_addr`][self::map_addr].
    |                           ^^^^^^^^^^^^^^^ no item named `with_addr` in module `const_ptr`
    |
    = note: `-D rustdoc::broken-intra-doc-links` implied by `-D warnings`
```

Reply:
this is the worst

Reply:
Shouldn't it be `Self` with a capital S?

Reply:
... 🤦🏻

yes, it should

Reply:
ngl i just figured you knew the forbidden rustdoc syntax lol

Reply:
no I was just discussing language internals while forgetting basic rust syntax

Reply:
meanwhile i'm proposing a whole new memory model and refactoring all of the stdlib while being stymied by the documentation link checker

Reply:
honestly at this point I'm just relieved to know this file is genuinely cursed and I'm not just the asshole here.

Docs failed again lol.

Reply:
tbf with `Self` not working I'm kind of convinced it's just a rustdoc bug that can be ignored for now

Reply:
I think pointer (primitive) might be a genuinely unique Hack inside rustdoc because it's both a magic primitive page *and* merging two types onto one page. I am trying one more thing and then I'll give up and delete the links (the ptr module itself uses the magic path pointer::offset, so perhaps that is the Right incantation).